documentation/asciidoc/computers/raspberry-pi/boot-http.adoc

```diff
@@ -40,6 +40,79 @@ rpi-eeprom-digest -i pieeprom.upd -o pieeprom.sig
 rpi-eeprom-digest -i boot.img -o boot.sig -k myprivkey.pem
 ----
 
+=== Certificates
+
+For security, Network Install uses HTTPS to download OS images from the Raspberry Pi website. This feature uses our own CA root included in the bootloader to verify the host.
+
+You can add your own custom CA certificate to the EEPROM to securely download images from your own website. Use the `--cacertder` option of the `rpi-eeprom-config` tool to add the DER-encoded certificate.
+
+To ensure that the certificate is not modified, place a hash of the certificate in the EEPROM config settings.
+
+Run the following command to generate a DER-encoded certificate:
+
+[source,console]
+----
+$ openssl x509 -in your_ca_root_cert.pem -out cert.der -outform DER
+----
+
+Then, run the following command to generate a SHA-256 hash of the certificate:
+
+[source,console]
+----
+$ sha256sum cert.der
+----
+
+You should see output similar to the following:
+
+----
+701bd97f67b0f5483a9734e6e5cf72f9a123407b346088638f597878563193fc  cert.der
+----
+
+Next, update `boot.conf` to include the hash of the certificate:
+
+[source,console]
+----
+$ sudo rpi-eeprom-config --edit
+----
+
+Configure the following settings in the `[gpio8=0]` section, replacing:
+
+* `<your_website>` with your website, e.g. `raspberrypi.com`
+* `<path_to_image>` with the path to your OS image hosted on your website, e.g. `path/to/image`
+* `<hash>` with the hash value you generated above, e.g. `701bd97f67b0f5483a9734e6e5cf72f9a123407b346088638f597878563193fc`
+
+----
+[all]
+BOOT_UART=1
+POWER_OFF_ON_HALT=0
+BOOT_ORDER=0xf461
+
+[gpio8=0]
+BOOT_ORDER=0xf7
+NET_INSTALL_ENABLED=0
+HTTP_HOST=<your_website>
+HTTP_PATH=<path_to_image>
+HTTP_CACERT_HASH=<hash>
+----
+
+NOTE: If you specify a `HTTP_CACERT_HASH`, Network Install downloads the image using HTTPS over port 443.
+
+Finally, use to load everything into EEPROM:
+
+[source,console]
+----
+$ rpi-eeprom-config -c boot.conf -p public_key.pem -o pieeprom.bin --cacertder cert.der pieeprom.original.bin
+$ rpi-eeprom-digest -k private_key.pem -i pieeprom.bin -o pieeprom.sig
+----
+
+During network boot, your Raspberry Pi should use HTTPS instead of HTTP. You can check the boot output for the HTTPS URL:
+
+----
+Loading boot.img ...
+HTTP: GET request for https://yourserver.org:443/path/to/files/boot.sig
+HTTP: GET request for https://yourserver.org:443/path/to/files/boot.img
```

Comment:
This may well be a stupid question (because I'm tired), but how does the firmware know to download both `https://yourserver.org:443/path/to/files/boot.sig` and `https://yourserver.org:443/path/to/files/boot.img` ?
Is `HTTP_PATH` in the eeprom-config set to `path/to/files/boot.img` (with the firmware then doing an extension-swap to get `path/to/files/boot.sig`); or is `HTTP_PATH` set to `path/to/files` (with this then meaning that your hosted OS image **has** to be named `boot.img`) ?

Reply:
Ah, https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#HTTP_HOST and https://www.raspberrypi.com/documentation/computers/raspberry-pi.html#HTTP_PATH answer my question (i.e. it's the latter), so perhaps this needs to be made a little bit clearer in this page?